enhancements/cluster-capabilities-api.md

```diff
@@ -0,0 +1,328 @@
+---
+title: cluster-capabilities-api
+authors:
+  - "@dhellmann"
+  - "@eranco"
+  - "@mrunalp"
+  - "@romfreiman"
+reviewers:
+  - TBD
+  - "@markmc"
+  - "@deads2k"
+  - "@wking"
+  - "@eparis"
+  - "@hexfusion"
+approvers:
+  - TBD
+creation-date: 2020-12-02
+last-updated: 2020-12-02
+status: implementable
+see-also:
+  - "/enhancements/single-node-production-edge-cluster-profile.md"
+  - "/enhancements/update/cluster-profiles.md"
+---
+
+# Cluster Capabilities API
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+As we add more deployment topologies for OpenShift, we need a way to
+communicate expectations to the operators managing the cluster. This
+enhancement describes an API for expressing the expected
+"capabilities" of a cluster, including aspects such as how hard
+operators should work to provide high-availability and whether
+operators should configure their operands in a way to minimize
+resource consumption for especially constrained environments.
+
+## Motivation
+
+We want to avoid having cluster profiles use different manifests to
+configure operators managed by the cluster-version-operator, and
+instead rely on in-cluster APIs. This approach will make the same
+information available to operators managed by either the
+cluster-version-operator or operator-lifecycle-manager.
+
+We do not want to expose the cluster profile name directly to most of
+the operators, because we do not want to have to update operators each
+time we add a new profile.
+
+We want the settings to be global to the cluster, rather than using
+operator-specific APIs, because that simplifies the integration with
+the installer and allows them to be applied consistently in all
+operators that need them.
+
+### Goals
+
+- Expose capability expectations to operators running in a cluster so
+  they can configure their operands appropriately.
+
+### Non-Goals
+
+- Expose cluster profile names to operators.
+- Provide a mutable API for changing the cluster capabilities (at
+  least for now).
```

Comment:
It might be worth a ```go section in the proposal with the content you expect to add to openshift/api for this.  That would make it clear that your new properties were going to appear in a less-mutable `status` fields vs. user-configurable `spec` fields.  Precedent for using `status` properties for write-once fields in the Infrastructure object.

Reply:
@wking I'm not sure what you mean here. Are you asking for a go struct definition?

Reply:
I think the immutability is covered with the prose now. Let me know if it's still not clear and we can do more.